### PR TITLE
detailed annotation for black formatting issue

### LINKED ISSUE #6003: Show more actionable annotation for python formatting issues
We use [problem matchers](https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md) to surface python code formatting issues detected by `black` on GitHub UI as shown in the screenshot below:

Screen Shot 2022-05-31 at 17 05 14

We want to replace `would reformat` with a more actionable message. To achieve this, we need to make the following changes:

1. Modify the output of `black` with `sed`:

```diff
diff --git a/dev/lint.sh b/dev/lint.sh
index 972bffe32..dc26a0a9e 100755
--- a/dev/lint.sh
+++ b/dev/lint.sh
@@ -7,14 +7,7 @@ trap 'err=1' ERR
-black --check .
-
-if [ $? -ne 0 ]; then
-  echo '
-To apply black foramtting, do one of the following:
-- Run `pip install $(cat requirements/lint-requirements.txt | grep "^black==") && black .`
-- Comment `autoformat` on the PR'
-fi
+black --check . 2>&1 | sed 's/would reformat/Run `black .` or comment `autoformat` on the PR to format this file:/g'
```

With this change, the output of `black` should look like this:

```bash
Skipping .ipynb files as Jupyter dependencies are not installed.
You can fix this by running ``pip install black[jupyter]``
Run `black .` or comment `autoformat` on the PR to format this file: tests/pyfunc/test_spark.py

Oh no! 💥 💔 💥
1 file would be reformatted, 608 files would be left unchanged.

# Original output:
# Skipping .ipynb files as Jupyter dependencies are not installed.
# You can fix this by running ``pip install black[jupyter]``
# would reformat tests/pyfunc/test_spark.py
#
# Oh no! 💥 💔 💥
# 1 file would be reformatted, 608 files would be left unchanged.
```

2. Fix `/.github/workflows/matchers/black.json` to capture the new message.

```diff
diff --git a/.github/workflows/matchers/black.json b/.github/workflows/matchers/black.json
index 605f10c27..aab652465 100644
--- a/.github/workflows/matchers/black.json
+++ b/.github/workflows/matchers/black.json
@@ -5,7 +5,7 @@
       "severity": "error",
       "pattern": [
         {
-          "regexp": "^(would reformat) (.+)$",
+          "regexp": "^(Run `black \.`.+): (.+)$",
           "file": 2,
           "message": 1
         }
```

If you want to work on this task, please ping me first.

### PR DESCRIPTION
**Pull request by subramaniam02**
Signed-off-by: subramaniam02 <subramaniam02@gmail.com>

## Related Issues/PRs

Close #6003 

## What changes are proposed in this pull request?

Improved Annotation message for black formatting issue

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by subramaniam02** [comment]
@harupy I am thinking to introduce a formatting issue in one of the python files and reverting the change to test. Will you `squash & merge` so that this change does not show up in commit history ? Or do you think tests is not needed for this change. Thanks !

**Comment by harupy** [comment]
> @harupy I am thinking to introduce a formatting issue in one of the python files and reverting the change to test. Will you `squash & merge` so that this change does not show up in commit history ? Or do you think tests is not needed for this change. Thanks !

Sounds good, let's do it :)

File: .github/workflows/matchers/black.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
       "severity": "error",
       "pattern": [
         {
-          "regexp": "^(would reformat) (.+)$",
+          "regexp": "^(Run `black \.`.+): (.+)$",
```
Comment (harupy) [inline .github/workflows/matchers/black.json:8]:
```suggestion
          "regexp": "^(Run `black \\.`.+): (.+)$",
```

Do we need one more `\`?

File: dev/lint.sh [hunk 2]
```diff
@@ -7,14 +7,8 @@ trap 'err=1' ERR
 
 echo -e "\n========== black ==========\n"
 # Exclude proto files because they are auto-generated
-black --check .
+black --check . 2>&1 | sed 's/would reformat/Run `black .` or comment `autoformat` on the PR to format this file: /g'
```
Comment (harupy) [inline dev/lint.sh:10]:
```suggestion
black --check . 2>&1 | sed 's/would reformat/Run `black .` or comment `autoformat` on the PR to format this file:/g'
```

**Comment by harupy** [comment]
Looks good to me once the comments are addressed and the manual test you described in https://github.com/mlflow/mlflow/pull/6023#issuecomment-1145334825 is done :)

**Comment by harupy** [comment]
Works as expected:

image

**Review by harupy — APPROVED** [review]
LGTM :)

**Comment by subramaniam02** [comment]
@harupy Sorry for the late response. I was keeping an eye on this PR but unfortunately I did not receive any notifications. Thanks for closing the PR, feel bad to not get it done till the end.

**Comment by harupy** [comment]
@subramaniam02 Not a problem :)

**Pull request merged**